### PR TITLE
support unix socket connection for influx CLI

### PR DESCRIPTION
**Pull request by oiooj**
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA]

How to use:

```
Sosara@MBA:~/influx$ ./influx -socket /tmp/influxdb.sock
Visit https://enterprise.influxdata.com to register for updates, InfluxDB server management, and monitoring.
Connected to /tmp/influxdb.sock version unknown
InfluxDB shell version: unknown
> show databases;
name: databases
name
----
_internal
foo

> 
```

and the default HTTP:

```
Sosara@MBA:~/influx$ ./influx 
Visit https://enterprise.influxdata.com to register for updates, InfluxDB server management, and monitoring.
Connected to http://localhost:8086 version unknown
InfluxDB shell version: unknown
> show databases;
name: databases
name
----
_internal
foo

>

```

### CODE REVIEW

**Comment by oiooj** [comment]
@jsternberg

File: cmd/influx/main.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ func main() {
 	fs := flag.NewFlagSet("InfluxDB shell version "+version, flag.ExitOnError)
 	fs.StringVar(&c.Host, "host", client.DefaultHost, "Influxdb host to connect to.")
 	fs.IntVar(&c.Port, "port", client.DefaultPort, "Influxdb port to connect to.")
+	fs.StringVar(&c.UnixSocket, "unixsocket", c.UnixSocket, "Influxdb unix socket to connect to.")
```
Comment (corylanou) [inline cmd/influx/main.go]: Can we change `unixsocket` to `socket`?  Otherwise I think this looks great!

File: cmd/influx/cli/cli.go [hunk 2]
```diff
@@ -318,7 +320,7 @@ func (c *CommandLine) Connect(cmd string) error {
 
 	var v string
 	if _, v, e = c.Client.Ping(); e != nil {
-		return fmt.Errorf("Failed to connect to %s\n", c.Client.Addr())
+		return fmt.Errorf("Failed to connect to %s: %s\n", c.Client.Addr(), e.Error())
```
Comment (e-dard) [inline cmd/influx/cli/cli.go]: This error message should begin with `failed` rather than `Failed`

File: cmd/influx/cli/cli.go [hunk 3]
```diff
@@ -133,8 +134,8 @@ func (c *CommandLine) Run() error {
 
 	if err := c.Connect(""); err != nil {
 		return fmt.Errorf(
-			"Failed to connect to %s\nPlease check your connection settings and ensure 'influxd' is running.",
-			c.Client.Addr())
+			"Failed to connect to %s: %s\nPlease check your connection settings and ensure 'influxd' is running.",
```
Comment (e-dard) [inline cmd/influx/cli/cli.go]: This error message should begin with `failed` rather than `Failed`.

**Comment by oiooj** [comment]
@corylanou updated

**Pull request merged**